docs/reference/indices/create-data-stream.asciidoc

```diff
@@ -0,0 +1,73 @@
+[[indices-create-data-stream]]
+=== Create data stream API
+++++
+<titleabbrev>Create data stream</titleabbrev>
+++++
+
+Creates a new data stream.
+
+Data streams provide a convenient way to ingest, search, and manage time series data. Documents
+containing time series data may be indexed directly into a data stream and searched through the
+data stream. Behind the scenes, data streams contain one or more hidden backing indices and a generation
+attribute that indicates which of the backing indices is the write index, the index into which documents
+will be ingested.
+
+Backing indices are always named `<data-stream-name>-zzzzzz` where `zzzzzz` is the six-digit, zero-padded
```

Comment:
I think we will violate this naming convention once we can shrink an index in a data stream (shrink in ilm creates a new index named `"shrink-<index-name>"` and since we cannot rename, we are likely stuck with that name. 

So we should loosen the wording here a bit (always at least seems too strong).